File: MoodByte/src/main/java/org/example/ModeloDTO/RegistroDTO.java
package org.example.ModeloDTO;

import org.example.Modelo.Registro;

import java.time.LocalDate;

public record RegistroDTO(
        int puntuacion,
        LocalDate fechaRegistro,
        long idUsuario
        ) {
    public RegistroDTO(Registro registro){
        this(
          registro.getPuntuacion(),
          registro.getFechaRegistro(),
          registro.getUsuario().getId()
        );
    }
}

File: MoodByte_App/app/src/main/java/com/example/moodbyte/data/remote/dtos/RegistroDto.kt
package com.example.moodbyte.data.remote.dtos

import com.example.moodbyte.data.local.entities.RegistroEntity
import java.time.LocalDate

data class RegistroDto(
    val puntuacion: Int,
    val fechaRegistro: String,
    val idUsuario: Long?
)
fun RegistroDto.toEntity() = RegistroEntity(
    puntuacion = puntuacion,
    fechaRegistro = fechaRegistro,
    idUsuario = idUsuario
)
